sql/vector_mhnsw.cc

```diff
@@ -735,7 +766,7 @@ int mhnsw_first(TABLE *table, KEY *keyinfo, Item *dist, ulonglong limit)
   *(ulonglong*)context= limit;
   context+= context_size;
 
-  while (limit--)
+  while (candidates.elements && limit--)
```

Comment:
not needed, I've done this few days ago with
{code:c}
  if (limit > candidates.num)
    limit= candidates.num;
{code}